nova_vm/src/heap/element_array.rs

```diff
@@ -2304,6 +2304,81 @@ impl ElementArrays {
                 .contains(&Some(element)),
         }
     }
+
+    /// This method creates a "shallow clone" of the elements of a trivial/dense array.
+    /// It does not do anything with descriptors and assumes there is a previous validation in place.
+    pub fn shallow_clone(&mut self, elements_vector: ElementsVector) -> SealableElementsVector {
+        let new_len = usize::try_from(elements_vector.len()).unwrap();
+        let new_key = ElementArrayKey::from(new_len);
+        let ElementArrays {
+            e2pow4,
+            e2pow6,
+            e2pow8,
+            e2pow10,
+            e2pow12,
+            e2pow16,
+            e2pow24,
+            e2pow32,
+        } = self;
+        let new_index = match new_key {
+            ElementArrayKey::Empty => ElementIndex::from_u32_index(0),
+            ElementArrayKey::E4 => {
+                let elements = e2pow4;
+                let index = ElementIndex::last_element_index(&elements.values).into_index();
```

Comment:
**issue:** This isn't correct; you're now always making a shallow copy of the last allocated elements in this particular `e2powX` array. What you want to do is to use the index from the `elements_vector` you took as a parameter, so that you make a shallow copy of that instead.